src/dscanner/analysis/ifelsesame.d

```diff
@@ -20,69 +16,60 @@ import dsymbol.scope_ : Scope;
  * $(LI == expressions where the left and right are the same)
  * )
  */
-final class IfElseSameCheck : BaseAnalyzer
+extern (C++) class IfElseSameCheck(AST) : BaseAnalyzerDmd
 {
-	alias visit = BaseAnalyzer.visit;
-
+	alias visit = BaseAnalyzerDmd.visit;
 	mixin AnalyzerInfo!"if_else_same_check";
 
-	this(string fileName, const(Scope)* sc, bool skipTests = false)
+	extern (D) this(string fileName, bool skipTests = false)
 	{
-		super(fileName, sc, skipTests);
+		super(fileName, skipTests);
 	}
 
-	override void visit(const IfStatement ifStatement)
+	override void visit(AST.IfStatement s)
 	{
-		if (ifStatement.thenStatement && (ifStatement.thenStatement == ifStatement.elseStatement))
-			addErrorMessage(ifStatement.line, ifStatement.column,
-					"dscanner.bugs.if_else_same", "'Else' branch is identical to 'Then' branch.");
-		ifStatement.accept(this);
-	}
+		if (s.elsebody && to!string(s.ifbody.toChars()) == to!string(s.elsebody.toChars()))
```

Comment:
I think that using `strcmp` is better than going through the hurdles of using `to`.

Reply:
Is strcmp behaviour identical to C? If so, wouldn't that leave the program vulnerable? Wouldn't it be better to use D's strings instead of char * ?